Correct Thermal Comfort Clothing Value Reporting Error

RKStrand opened the pull request:
Pull request overview
---------------------

- Fixes #11120

### Description of the purpose of this PR

A user discovered that when selecting scheduled clothing values that the reported value for the Fanger model was always zero.  There were various missing assignments that caused this to be the case for several thermal comfort models, always when the clothing value was schedule driven only.  This did not affect the results because internally the calculation variables were updated properly.  It only affected the report value.  This issue has been corrected and now the clothing value is reported properly.  A unit test created for this fix verifies this.  The variable in question was actually not even listed in the Input Output Reference even though the code clearly made this a valid report variable.  The documentation was updated to include this variable as an option.  In addition, while working on this defect, it was discovered that two input fields for the People object were not documented in the Input Output Reference.  This was also corrected.  Finally, an IDF in the test suite was modified to request the report variable that was corrected since no IDF files in the test suite requested it.

### Pull Request Author

<!-- Add to this list or remove from it as applicable.  This is a simple templated set of guidelines. -->

 - [X] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [X] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [X] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer

<!-- This will not be exhaustively relevant to every PR. -->

 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

RKStrand committed:
Correction to Populate Thermal Comfort Output Variable

The current clothing value was not being reported properly for situations when the user had clothing insulation scheduled.  This was an issue for several for several models.  This corrects the problem and the output value is properly assigned rather than always zero.

RKStrand committed:
Missed Pierce Model, Unit Test, Docs Changes

The Pierce model also did not have any code to set the clothing value reporting variable.  This corrects this.  Also a unit test verifying the fix was added.  Finally, this variable was officially added to the IO Ref and two missing input fields were documented as well.

RKStrand committed:
Correction of "missing" documentation

Appears that the two fields that had "missing" desriptions were not missing but rather misplaced.  Moved the information to the correct place and added some degree symbols that were missing throughout this section.

mjwitte committed:
Merge remote-tracking branch 'remotes/origin/develop' into 11120CloInsulationReportingIssue

mjwitte committed:
Add new output variable to rvi

mjwitte reviewed (approved):
@RKStrand OK, tested this, works as advertised. I merged in develop and added the new output to the rvi file. Will wait for CI before merging.

Myoldmopar commented on doc/input-output-reference/src/overview/group-internal-gains-people-lights-other.tex:
```diff
@@ -132,22 +132,6 @@ \subsubsection{Inputs}\label{inputs-025}
-is defined as the longest duration (number of hours), starting from the
-beginning time of the risk period (e.g., the start time of a power outage), to
-not above a certain temperature threshold (specified in this field). If not
-specified, 86F (or 30C) will be used as the default temperature threshold.
```
These were moved/removed?  I didn't expect that from the PR text.  Could just be a groggy start for me though :)

RKStrand replied:
Yes, they were moved because the order did not match the IDD before.  In the current IDD, these are placed at the end (N7, N8) of the object syntax.  I moved them to match the IDD.

Myoldmopar commented on doc/input-output-reference/src/overview/group-internal-gains-people-lights-other.tex:
```diff
@@ -301,6 +285,22 @@ \subsubsection{Inputs}\label{inputs-025}
+is defined as the longest duration (number of hours), starting from the
+beginning time of the risk period (e.g., the start time of a power outage) to
+not above a certain temperature threshold (specified in this field). If not
+specified, 86$^\circ$F (or 30$^\circ$C) will be used as the default temperature threshold.
```
So these were moved down here?  It kinda feels a bit odd given the paragraph text above saying the IDF text follows.

RKStrand replied:
My bad.  I moved that paragraph and made another commit.  Should be better now.

Myoldmopar commented on src/EnergyPlus/ThermalComfort.cc:
```diff
@@ -1506,6 +1509,7 @@ namespace ThermalComfort {
             switch (people.clothingType) {
             case DataHeatBalance::ClothingType::InsulationSchedule: {
                 state.dataThermalComforts->CloUnit = people.clothingSched->getCurrentVal();
+                comfort.ClothingValue = state.dataThermalComforts->CloUnit;
```
Got it, so every scheduled block you need to update it.

Myoldmopar reviewed (commented):
I'm a little surprised with the move in the docs.  Is the placement correct?  It might be and I'm just missing something.  Anyway, the functional changes are great so if you can confirm the doc comment, we can get this merged.

RKStrand committed:
Misplace paragraph fix

Docs fix--when sections moved, a paragraph that leads into the next section was not accounted for.

Myoldmopar committed:
Merge remote-tracking branch 'origin/develop' into 11120CloInsulationReportingIssue

Myoldmopar reviewed (approved):
Thanks for addressing my comments, @RKStrand.  I'm doing a quick build locally.  Looks like there is an unused variable, which is now properly causing a build error.  So I'll fix that, push, and let CI give me one sanity check.

Myoldmopar committed:
Fix unused variable in unit test
